engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1183,7 +1215,11 @@ public void removeContainerAgent(String containerId)
     if (containerAgent != null) {
       // record operator stop for this container
       for (PTOperator oper : containerAgent.container.getOperators()) {
-        StramEvent ev = new StramEvent.StopOperatorEvent(oper.getName(), oper.getId(), containerId);
+        long groupId = 0;
+        if (redeployRequests.get(containerId) != null) {
+          groupId = redeployRequests.get(containerId).getGroupId();
+        }
+        StramEvent ev = new StramEvent.StopOperatorEvent(oper.getName(), oper.getId(), containerId, groupId);
```

Comment:
Shouldn't you also be removing entry for containerId from redeployRequests at this point?